### PR TITLE
Feat: get all sessions and session by id endpoint

### PR DESCRIPTION
**Pull request by manthanguptaa**
Merge this first in #1133

### CODE REVIEW

File: phi/playground/playground.py [hunk 1]
```diff
@@ -136,6 +136,20 @@ def agent_chat(body: AgentRunRequest):
                 run_response = cast(RunResponse, agent.run(body.message, stream=False))
                 return run_response.model_dump_json()
 
+        @playground_routes.post("/agent/sessions/all")
```
Comment (ashpreetbedi) [inline phi/playground/playground.py:139]: @manthanguptaa 
1. Let's add a `body: GetAgentSessionsRequest` that includes an `agent_id` and `user_id`
2. Let's get all sessions from storage using `agent.storage.get_all_sessions(user_id=user_id)`
3. Let's return `session_id` and `session_data` only

So the code will look like:
```
agent: Optional[Agent] = None
for _agent in self.agents:
    if _agent.agent_id == body.agent_id:
        agent = _agent
        break

if agent.storage is None:
    return JSONResponse(status=404, content=f"Agent does not have storage enabled.")

agent_sessions: List[GetAgentSessions] = []
all_agent_sessions: List[AgentSession] = agent.storage.get_all_sessions(user_id=user_id)
for session in all_agent_sessions:
      agent_sessions.append(GetAgentSessions(
          session_id=session.session_id,
          session_data=session.session_data,
      ))
return sessions
```

File: phi/playground/playground.py [hunk 2]
```diff
@@ -136,6 +136,20 @@ def agent_chat(body: AgentRunRequest):
...
+        def get_agent_sessions():
+            session_data = []
+            for agent in self.agents:
+                session_data.append(agent.get_session_data())
+            return session_data
+
+        @playground_routes.post("/agent/sessions/{session_id}")
```
Comment (ashpreetbedi) [inline phi/playground/playground.py:146]: @manthanguptaa 
1. Let's add a `body: ReadAgentSessionRequest` that includes an `agent_id` and `user_id`
2. Let's make a call to `.read_from_storage()` to read the data from storage first
3. Let's return the AgentSession object

So the code will look like:
```
agent: Optional[Agent] = None
for _agent in self.agents:
    if _agent.agent_id == body.agent_id:
        agent = _agent(session_id=session_id)
        break

agent.read_from_storage()
return agent.to_agent_session()
```

**Review by ashpreetbedi — CHANGES_REQUESTED** [review]

File: phi/playground/playground.py (hunk 2 above)
Comment (manthanguptaa) [reply phi/playground/playground.py:146]: agent = _agent(session_id=session_id) 
why are we making a call like this?

Comment (ashpreetbedi) [reply phi/playground/playground.py:146]:
```
agent: Optional[Agent] = None
for _agent in self.agents:
    if _agent.agent_id == body.agent_id:
        agent = _agent
        break

agent.session_id = session_id
agent.read_from_storage()
return agent.to_agent_session()
```

File: phi/playground/playground.py [hunk 3]
```diff
@@ -136,6 +147,46 @@ def agent_chat(body: AgentRunRequest):
...
+        @playground_routes.post("/agent/sessions/all")
+        def get_agent_sessions(body: GetAgentSessionsRequest):
+            agent: Optional[Agent] = None
+            for _agent in self.agents:
+                if _agent.agent_id == body.agent_id:
+                    agent = _agent
+                    break
+
+            if agent is None:
+                return JSONResponse(status_code=404, content=f"couldn't find agent with id {body.agent_id}")
```
Comment (ashpreetbedi) [inline phi/playground/playground.py:159]: @manthanguptaa JSONResponse(status_code=404, content="Agent not found.")

**Review by ashpreetbedi — APPROVED** [review]

**Pull request merged**